src/modules/hunting/etcd.py

```diff
@@ -81,66 +50,57 @@ def db_keys_write_access(self):
 
     def db_keys_disclosure(self):
-        logging.debug(self.event.host)
-        logging.debug("Passive hunter is attempting to read etcd keys remotely")
-        r_secure = "https://{host}:{port}/v2/keys".format(host=self.event.host, port=2379)
-        r_not_secure = "http://{host}:{port}/v2/keys".format(host=self.event.host, port=2379)
-        res = helperFuncDo2Requests(r_secure, r_not_secure)
-        if res:
-            self.keys_evidence = res.content
-            return True
-        return False
+        logging.debug(self.event.host + " Passive hunter is attempting to read etcd keys remotely")
+        r = requests.get(
+            "{protocol}://{host}:{port}/v2/keys".format(protocol=self.protocol, host=self.event.host, port=2379),
+            verify=False)
+        self.keys_evidence = r.content if r.status_code == 200 and r.content != '' else False
+        return self.version_evidence
 
     def version_disclosure(self):
-        logging.debug(self.event.host)
-        logging.debug("Passive hunter is attempting to check etcd version remotely")
-        r_secure = "https://{host}:{port}/version".format(host=self.event.host, port=2379)
-        r_not_secure = "http://{host}:{port}/version".format(host=self.event.host, port=2379)
-        res = helperFuncDo2Requests(r_secure, r_not_secure)
-        if res:
-            self.version_evidence = res.content
-            return True
-        return False
+        logging.debug(self.event.host + " Passive hunter is attempting to check etcd version remotely")
+        r = requests.get(
+            "{protocol}://{host}:{port}/version".format(protocol=self.protocol, host=self.event.host, port=2379),
+            verify=False)
+        self.version_evidence = r.content if r.status_code == 200 and r.content != '' else False
+        return self.version_evidence
 
     def unauthorized_access(self):
-        logging.debug(self.event.host)
-        logging.debug("Passive hunter is attempting to access etcd without authorization")
-        r_not_secure = "http://{host}:{port}/version".format(host=self.event.host, port=2379)
-        res = helperFuncDo2Requests(r_not_secure, r_not_secure)  # We don't have to do 2 requests this time
-        if res:
-            return True
-        return False
+        logging.debug(self.event.host + " Passive hunter is attempting to access etcd without authorization")
+        r = requests.get("http://{host}:{port}/version".format(host=self.event.host, port=2379), verify=False)
+        return r.content if r.status_code == 200 and r.content != '' else False
 
     def execute(self):
+        if self.unauthorized_access():  # decide between http and https protocol
+            self.protocol = 'http'
         if self.version_disclosure():
             self.publish_event(EtcdRemoteVersionDisclosureEvent(self.version_evidence))
-            if self.unauthorized_access():
+            if self.protocol == 'http' and self.unauthorized_access():
```

Comment:
I think you could publish the event after the first time you try self.unauthorized_access() a few lines above (and if it succeeds set http and publish the event at the same time)

Reply:
The event is indeed logically can be published there, but we won't have the evidence for it.
we can OFC change the unauthorized_access() to save version evidence on success but it would make the method less simple so I have chose to publish it after we have received the evidence.

That being said, I think we can make that logic a bit niter:
`    def execute(self):
        if self.insecure_access():  # make a decision between http and https protocol
            self.protocol = 'http'
        if self.version_disclosure():
            self.publish_event(EtcdRemoteVersionDisclosureEvent(self.version_evidence))
            **if self.protocol == 'http':  # If protocol is HTTP, self.insecure_access() had succeeded so we dont have to do the same request again***
                self.publish_event(EtcdAccessEnabledWithoutAuthEvent(self.version_evidence))
            if self.db_keys_disclosure():
                self.publish_event(EtcdRemoteReadAccessEvent(self.keys_evidence))
`